src/main/java/com/venafi/vcert/sdk/connectors/tpp/TppTokenConnector.java

```diff
@@ -412,20 +435,44 @@ public String renewCertificate(RenewalRequest request, String accessToken) throw
 
 
     @Override
-    public ImportResponse importCertificate(ImportRequest request, String accessToken) throws VCertException {
+    public ImportResponse importCertificate(ImportRequest request) throws VCertException {
         if (isBlank(request.policyDN())) {
             request.policyDN(getPolicyDN(zone));
         }
 
-        return doImportCertificate(request, accessToken);
+        return doImportCertificate(request);
     }
 
-    private ImportResponse doImportCertificate(ImportRequest request, String accessToken) throws VCertException {
-        return tpp.importCertificateToken(request, getAuthHeaderValue(accessToken));
+    private ImportResponse doImportCertificate(ImportRequest request) throws VCertException {
+        return tpp.importCertificateToken(request, getAuthHeaderValue());
     }
 
     @Override
-    public Policy readPolicyConfiguration(String zone, String accessToken) throws VCertException {
+    public Policy readPolicyConfiguration(String zone) throws VCertException {
         throw new UnsupportedOperationException("Method not yet implemented");
     }
+
+    private boolean isEmptyCredentials(Authentication credentials){
```

Comment:
should we check for api key here too?

Reply:
The TppTokenConnector class is intended to be used specifically for the TPP with the api operations that support token header. It is not expected that the api key is present in the credentials, nor it is needed for the use case.